packages/notifier/src/notifier.js

```diff
@@ -155,11 +155,12 @@ export const makeNotifierKit = (...args) => {
  * Adaptor from async iterable to notifier.
  *
  * @template T
- * @param {AsyncIterable<T>} asyncIterable
+ * @param {ERef<AsyncIterable<T>>} asyncIterableP
  * @returns {Notifier<T>}
  */
-export const makeNotifierFromAsyncIterable = asyncIterable => {
-  const iterator = asyncIterable[Symbol.asyncIterator]();
+export const makeNotifierFromAsyncIterable = asyncIterableP => {
+  /** @type {ERef<AsyncIterator<T>>} */
```

Comment:
`makeNotifierFromAsyncIterable` has a lot of the internal type decl comments that @dtribble  rightly point out we should minimize. I wrote most of them, but this PR adds yet another. I just tried the first obvious thing of removing each, and found that this new one is necessary, but the declarations on `optNextPromise`, `currentResponse`, and `baseNotifier` can be omitted without causing a failure to type check.

You are the master of jsdoc type checking and properjs, so I leave this to your judgement. No change needed as of this PR, especially since the new one as of this PR is necessary. But perhaps `makeNotifierFromAsyncIterable` will be an interesting case study of this issue going forward?

Reply:
I think that annotating a declaration is perfectly fine, and not excessive.

The "type goo" that Dean refers to is usually more than one line of gymnastics to get existing code to conform to a type declaration.  I agree, and try to avoid that where possible.

Reply:
@dtribble I'd appreciate your commentary, but in the meantime I'm resolving this conversation. @michaelfig since the type declarations at issue are not introduced by the PR anyway, do not await @dtribble 's commentary before merging.